### PR TITLE
PLT-4357 adding performance monitoring

### PR DESCRIPTION
**Pull request by coreyhulen**
#### Summary
Adding performance monitoring

Enterprise PR at https://github.com/mattermost/enterprise/pull/68

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-4357

#### Checklist
- [X] Added or updated unit tests (required for all new features)
- [X] Has enterprise changes (please link)
- [X] Has UI changes
- [X] Includes text changes and localization file ([.../i18n/en.json]

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -49,8 +57,12 @@ imports:
   - gf256
   - qr
   - qr/coding
+- name: github.com/matttproud/golang_protobuf_extensions
+  version: 3247c84500bff8d9fb6d579d800f20b3e091582c
+  subpackages:
+  - pbutil
 - name: github.com/miekg/dns
-  version: 58f52c57ce9df13460ac68200cef30a008b9c468
+  version: a8a1273f08d171f75254d2157ee546ea65bd0a27
```
Comment (hmhealey) [inline glide.lock]: For these packages that changed (github.com/miekg/dns, github.com/tylerb/graceful, and github.com/xenolf/lego), are you upgrading them or is this accidentally reverting when Chris updated them last week?

File: api/context.go [hunk 2]
```diff
@@ -228,13 +230,27 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 			if c.Err.StatusCode == http.StatusUnauthorized {
 				http.Redirect(w, r, c.GetTeamURL()+"/?redirect="+url.QueryEscape(r.URL.Path), http.StatusTemporaryRedirect)
 			} else {
 				RenderWebError(c.Err, w, r)
 			}
 		}
+
+	}
+
+	if h.isApi && einterfaces.GetMetricsInterface() != nil {
```
Comment (crspeller) [inline api/context.go]: Shouldn't we be checking to see if it's disabled here? Also in a lot of other places.

Comment (crspeller) [inline api/context.go]: Unless of course we always want to collect, then ignore this comment.

File: glide.lock (hunk 1 above)
Comment (crspeller) [inline glide.lock]: We shouldn't even be using graceful any more. It's probably a dependency of a dependency? Like the others you mentioned.

Comment (coreyhulen) [inline glide.lock]: I chatted with CS and HH.  I'll move everything to tylerb/graceful.

**Pull request merged**